src/interfaces/toMarcRecords.js

```diff
@@ -1,67 +1,86 @@
 import {Json, MARCXML, AlephSequential, ISO2709} from '@natlibfi/marc-record-serializers';
 import ConversionError, {Utils} from '@natlibfi/melinda-commons';
-import {amqpFactory, OPERATIONS, logError} from '@natlibfi/melinda-rest-api-commons';
-import {AMQP_URL} from '../config';
+import {OPERATIONS, logError} from '@natlibfi/melinda-rest-api-commons';
 import {updateField001ToParamId} from '../utils';
 
-const {createLogger} = Utils;
-
-export async function streamToMarcRecords({correlationId, operation, contentType, stream}) {
+export default async function (amqpOperator) {
+	const {createLogger} = Utils;
 	const logger = createLogger();
-	let recordNumber = 0;
-	let promises = [];
-	const reader = chooseAndInitReader(contentType);
-	const amqpOperator = await amqpFactory(AMQP_URL);
-
-	// Purge queue before importing records in
-	await amqpOperator.checkQueue(correlationId, 'messages', true);
-
-	await new Promise((resolve, reject) => {
-		reader.on('error', err => {
-			logError(err);
-			reject(new ConversionError(422, 'Invalid payload!'));
-		}).on('data', data => {
-			promises.push(transform(data));
-
-			async function transform(record) {
+
+	return {streamToRecords};
```

Comment:
No need to return an object but just a default function instead?

Reply:
with this i can pass amqpOperator separately to request + maintain just one connection to amqp and more unified code style